flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -563,7 +554,7 @@ private void startTriggeringCheckpoint(CheckpointTriggerRequest request) {
                                             createPendingCheckpoint(
                                                     timestamp,
                                                     request.props,
-                                                    ackTasks,
+                                                    checkpointBrief,
```

Comment:
What I understand is that the `CheckpointBrief` describes how to “get” a `CompleteChekcpoint`. There are some methods related to“get”a CompleteCheckpoint, such as 
“createPendingCheckpoint/reportToStateTracker/sendAbortedMessage/sendAcknowledgeMessages”.

However, these methods sometime use `CheckpointBrief`, sometime uses `tasksTowaitFor/tasksToTriggerFor`, which are the internal state of the `CheckpointCoordinator`. After we refactor maybe all the methods could only use the `CheckpointBrief` as their parameter. (I think CheckpointId,TimeStamps should also be in the CheckpointBrief).

Reply:
Very thanks @guoweiM  for the thoughts! For me I'm more tend to make function parameters more "specific" and only use the composite object if we relies on multiple properties of it. But I'm also open to other options. :-)

Reply:
I think both are good points, whatever we decide we could also change that in a later refactoring, right?

Reply:
I'll also +1 for implement it in later refactoring~